File: project/jimmer-core/src/test/java/org/babyfish/jimmer/model/RegisterRequest.java
package org.babyfish.jimmer.model;

import org.babyfish.jimmer.Immutable;

@SamePassword
@Immutable
public interface RegisterRequest {

    String name();

    String password();

    String passwordAgain();

    @IdCard(message = "{i18n.idCardError}")
    String idCard();
}

File: project/jimmer-core-kotlin/src/test/kotlin/org/babyfish/jimmer/kt/model/RegisterRequest.kt
package org.babyfish.jimmer.kt.model

import org.babyfish.jimmer.Immutable

@SamePassword
@Immutable
interface RegisterRequest {

    val name: String

    val password: String

    val passwordAgain: String

    @IdCard(message = "{i18n.idCardError}")
    val idCard: String
}